### PR TITLE
Remove internal Downloader abstraction.

### PR DESCRIPTION
**Pull request by NightlyNexus**

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso/NetworkRequestHandler.java [hunk 1]
```diff
@@ -28,10 +28,10 @@
   private static final String SCHEME_HTTP = "http";
   private static final String SCHEME_HTTPS = "https";
 
-  private final Downloader downloader;
+  private final OkHttp3Downloader downloader;
```
[the remark this answers is not in the document]
Comment (NightlyNexus) [reply picasso/src/main/java/com/squareup/picasso/NetworkRequestHandler.java]: yeah, i think we should. i'll follow this up.

File: picasso/src/test/java/com/squareup/picasso/TestUtils.java [hunk 2]
```diff
@@ -271,6 +274,54 @@ static Bitmap makeBitmap(int width, int height) {
     return Bitmap.createBitmap(width, height, ALPHA_8);
   }
 
+  static final OkHttp3Downloader UNUSED_DOWNLOADER = new OkHttp3Downloader(new Call.Factory() {
+    @Override public Call newCall(okhttp3.Request request) {
+      throw new AssertionError();
+    }
+  }, null, true);
+
+  static final class PremadeCall implements Call {
```
[the remark this answers is not in the document]
Comment (NightlyNexus) [reply picasso/src/test/java/com/squareup/picasso/TestUtils.java]: well, this test probably wanted to use MockWebServer, to be fair, but maybe.

[the remark this answers is not in the document]
Comment (JakeWharton) [reply picasso/src/test/java/com/squareup/picasso/TestUtils.java]: Yeah even better.

File: picasso/src/test/java/com/squareup/picasso/DispatcherTest.java [hunk 3]
```diff
@@ -64,13 +69,13 @@
...
 public class DispatcherTest {
-
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
   @Mock Context context;
   @Mock ConnectivityManager connectivityManager;
   @Mock PicassoExecutorService service;
   @Mock ExecutorService serviceMock;
   @Mock Handler mainThreadHandler;
-  @Mock Downloader downloader;
+  final OkHttp3Downloader downloader = UNUSED_DOWNLOADER;
```
Comment (jrodbx) [inline picasso/src/test/java/com/squareup/picasso/DispatcherTest.java]: nit: move below Mocks

File: picasso/src/test/java/com/squareup/picasso/DispatcherTest.java [hunk 4]
```diff
@@ -88,9 +93,17 @@
     verify(service).shutdown();
   }
 
-  @Test public void shutdownStopsDownloader() {
+  @Test public void shutdownShutsDownDownloader() {
+    dispatcher.shutdown();
+    okhttp3.Cache cache = new okhttp3.Cache(temporaryFolder.getRoot(), 100);
+    Dispatcher dispatcher = new Dispatcher(context, service, mainThreadHandler,
+        new OkHttp3Downloader(new Call.Factory() {
```
Comment (jrodbx) [inline picasso/src/test/java/com/squareup/picasso/DispatcherTest.java]: nit: extract downloader to var for readability

File: picasso/src/test/java/com/squareup/picasso/TestUtils.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrodbx) [reply picasso/src/test/java/com/squareup/picasso/TestUtils.java]: I like the fake Call concept.

**Comment by jrodbx** [comment]
https://github.com/square/picasso/commit/6f17d2df3e5339a318820f6eaec041f8f6dea1bb

**Pull request closed without merging**